float32 rad
float32 azi
float32 ele
int32 index